text/0000-const-wildcard.md

```diff
@@ -0,0 +1,120 @@
+- Feature Name: `const_wildcard`
+- Start Date: 2018-08-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow assigning constants to `_`, as in `const _: TYPE = VALUE`, analogous to
+`let _ = VALUE`.
+
+# Motivation
+[motivation]: #motivation
+
+The ability to ensure that code type checks while discarding the result is
+useful, especially in custom derives. For example, the following code will not
+compile if the type `MyType` doesn't implement the trait `MyTrait`:
+
+```rust
+const _FOO: () = {
+    use std::marker::PhantomData;
+    struct ImpelementsMyTrait<T: MyTrait>(PhantomData<T>);
+    let _ = ImplementsMyTrait::<MyType>(PhantomData); // type checking error if MyType: !MyTrait
+    ()
+};
+```
+
+Unfortunately, this requires coming up with a unique identifier to assign to.
+This is error-prone because no matter what identifier is chosen, there's always
+a possibility that a user will have already used the same identifier in their
+code. If writing `const _: () = { ... }` were valid, then this would be a
+non-issue - the `const _` could be repeated many times without conflicting with
+any other identifier in scope.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Allow assigning to `_` when defining a new constant. Just like `let _`, this
+doesn't introduce any new bindings, but still evaluates the rvalue at compile
+time like any other constant.
```

Comment:
A guide-level explanation does not include the word "rvalue", IMO. And this doesn't explain what the point of the feature is.

Reply:
Is there a phrase you'd suggest instead of "rvalue"?

The point of the feature is described in the "Motivation" section.

Reply:
I think we've called it `rvalue` for some time as well.. see https://github.com/rust-lang/rfcs/blob/master/text/1414-rvalue_static_promotion.md.

Reply:
We *used* to call it rvalue, but we've wisened up since then. Likewise with lvalues (now places).

Reply:
@rkruppe you speak the truth :)